cime/scripts/lib/CIME/SystemTests/homme.py

```diff
@@ -55,26 +78,60 @@ def run_phase(self):
+                    if os.path.isdir(full_baseline_fast_dir):
+                        shutil.rmtree(full_baseline_fast_dir)
+                    if os.path.isdir(full_baseline_strict_dir):
+                        shutil.rmtree(full_baseline_strict_dir)
+
+                    shutil.copytree(os.path.join(exeroot_fast, "tests", "baseline"), full_baseline_fast_dir)
+                    shutil.copytree(os.path.join(exeroot_strict, "tests", "baseline"), full_baseline_strict_dir)
+
+            elif compare:
+                stat = run_cmd("ctest -E 'cxx|limiters_ut|remap_ut|sphere_op_ut'", arg_stdout=log, combine_output=True, from_dir=exeroot_fast)[0]
+                # Only run the dev and unit profiles. This excludes all theta/sweqx/baro tests.
+                # That's what we want: we only want to test cxx vs f90 impls, so preqx vs preqx_kokkos and unit tests.
+                stat = stat + run_cmd("ctest -L 'dev|unit'", arg_stdout=log, combine_output=True, from_dir=exeroot_strict)[0]
 
-                shutil.copytree(os.path.join(exeroot, "tests", "baseline"), full_baseline_dir)
-
-        elif compare:
-            stat = run_cmd("{} -j 4 check".format(gmake), arg_stdout=log, combine_output=True, from_dir=exeroot)[0]
+            else:
+                stat = run_cmd("{} -j 4 baseline".format(gmake), arg_stdout=log, combine_output=True, from_dir=exeroot_fast)[0]
+                stat = stat + run_cmd("{} -j 4 baseline".format(gmake), arg_stdout=log, combine_output=True, from_dir=exeroot_strict)[0]
 
+                stat = run_cmd("ctest -E 'cxx|limiters_ut|remap_ut|sphere_op_ut'", arg_stdout=log, combine_output=True, from_dir=exeroot_fast)[0]
+                # Only run the dev and unit profiles. This excludes all theta/sweqx/baro tests.
+                # That's what we want: we only want to test cxx vs f90 impls, so preqx vs preqx_kokkos and unit tests.
+                stat = stat + run_cmd("ctest -L 'dev|unit'", arg_stdout=log, combine_output=True, from_dir=exeroot_strict)[0]
         else:
-            stat = run_cmd("{} -j 4 baseline".format(gmake), arg_stdout=log, combine_output=True, from_dir=exeroot)[0]
-            stat = run_cmd("{} -j 4 check".format(gmake), arg_stdout=log, combine_output=True, from_dir=exeroot)[0]
+            if generate:
+                full_baseline_dir = os.path.join(baseline, basegen, "tests", "baseline")
+                stat = run_cmd("{} -j 4 baseline".format(gmake), arg_stdout=log, combine_output=True, from_dir=exeroot)[0]
+                if stat == 0:
+                    if os.path.isdir(full_baseline_dir):
+                        shutil.rmtree(full_baseline_dir)
+
+                    shutil.copytree(os.path.join(exeroot_fast, "tests", "baseline"), full_baseline_dir)
+
+            elif compare:
+                stat = run_cmd("ctest -E 'cxx|limiters_ut|remap_ut|sphere_op_ut'", arg_stdout=log, combine_output=True, from_dir=exeroot)[0]
```

Comment:
Is this for any other machine than anvil/skb? If so, we probably want default behavior for now here? 'make check' instead of ctest?

Reply:
There are two issues: having or not having strict vs fast, and having or not having preqx_kokkos. So far the two go together: skb and anvil are the only testbeds with preqx_kokkos, and both have strict and fast.

However, one could enable preqx_kokkos on another machine file (providing a kokkos path). If so, the default would be to run a fast fp-model, which would fail all cxx_vs_f90 tests as well as the limiters_ut, remap_ut and sphere_op_ut ones.

I wanted to make sure that if someone activates preqx_kokkos (and generates baselines for it), tests still pass. Other than that, there would be no difference in make check vs ctest.

Reply:
I just worry that even though these changes look innocent, we cannot (or rather don't want to cause they are out of scope of this pr) test them. Tests are run on many machines and if someone really knows how to handle kokkos, they can use homme standalone build to test. Also, when this pr goes in, there may be surprises. If we don't change what we don't need to, we would have less to debug. Let's keep changes to other machines minimal for this pr, which is already enormous? thanks!

Reply:
Alright, I will push a fix.